demos/projects/ESPRESSIF/az-nvs-cert-bundle/main/az_nvs_cert_bundle_main.c

```diff
@@ -96,183 +97,216 @@ static uint8_t trust_bundle[] =
+
+    if( xESPErr != ESP_OK )
+    {
+        printf( "Error (%s) getting AZURE_TRUST_BUNDLE_VERSION_NAME from NVS!\n", esp_err_to_name( xESPErr ) );
+        return xESPErr;
+    }
 
-    /* Read AZURE_TRUST_BUNDLE_VERSION_NAME */
-    int32_t trust_bundle_read_version = 0; /* value will default to 0, if not set yet in NVS */
+    if( xTrustBundleVersionReadSize > 8 )
+    {
+        printf( "Not enough size to read version\n" );
+        return ESP_ERR_NO_MEM;
+    }
 
-    err = nvs_get_i32( my_handle, AZURE_TRUST_BUNDLE_VERSION_NAME, &trust_bundle_read_version );
+    /* Read the current trust bundle version */
+    xESPErr = nvs_get_blob( xNVSHandle, AZURE_TRUST_BUNDLE_VERSION_NAME, &ucReadTrustBundleVersion, &xTrustBundleVersionReadSize );
 
-    if( ( err != ESP_OK ) && ( err != ESP_ERR_NVS_NOT_FOUND ) )
+    if( xESPErr != ESP_OK )
     {
-        return err;
+        printf( "Error (%s) getting AZURE_TRUST_BUNDLE_VERSION_NAME from NVS!\n", esp_err_to_name( xESPErr ) );
+        return xESPErr;
     }
 
-    printf( "Current trust bundle version = %d\n", ( int ) trust_bundle_read_version );
+    printf( "Current trust bundle version = %.*s\n", xTrustBundleVersionReadSize, ucReadTrustBundleVersion );
 
     /* Get size of cert */
-    size_t trust_bundle_read_size = 0; /* value will default to 0, if not set yet in NVS */
-    err = nvs_get_blob( my_handle, AZURE_TRUST_BUNDLE_NAME, NULL, &trust_bundle_read_size );
+    xESPErr = nvs_get_blob( xNVSHandle, AZURE_TRUST_BUNDLE_NAME, NULL, &xTrustBundleReadSize );
 
-    if( ( err != ESP_OK ) && ( err != ESP_ERR_NVS_NOT_FOUND ) )
+    if( ( xESPErr != ESP_OK ) && ( xESPErr != ESP_ERR_NVS_NOT_FOUND ) )
     {
-        return err;
+        return xESPErr;
     }
 
-    if( trust_bundle_read_size == 0 )
+    if( xTrustBundleReadSize == 0 )
     {
         printf( "Nothing saved yet!\n" );
     }
     else
     {
-        uint8_t * trust_bundle_read = malloc( trust_bundle_read_size );
-        err = nvs_get_blob( my_handle, AZURE_TRUST_BUNDLE_NAME, trust_bundle_read, &trust_bundle_read_size );
+        uint8_t * trust_bundle_read = malloc( xTrustBundleReadSize );
+        xESPErr = nvs_get_blob( xNVSHandle, AZURE_TRUST_BUNDLE_NAME, trust_bundle_read, &xTrustBundleReadSize );
 
-        if( err != ESP_OK )
+        if( xESPErr != ESP_OK )
         {
             free( trust_bundle_read );
-            return err;
+            return xESPErr;
         }
 
         printf( "Stored trust bundle:\n" );
-        printf( "%.*s\n", trust_bundle_read_size, trust_bundle_read );
+        printf( "%.*s\n", xTrustBundleReadSize, trust_bundle_read );
 
         free( trust_bundle_read );
     }
 
-    return err;
+    return xESPErr;
 }
 
 /*
  *  Save the trust bundle (certs and version) to a namespace in the NVS
  */
-esp_err_t save_trust_bundle( nvs_handle_t my_handle )
+esp_err_t prvSaveTrustBundle( nvs_handle_t xNVSHandle )
```

Comment:
Same as the above comment - please remove this version and maintain only the Read/Write functions in the trust_bundle_storage  compilation unit.

Reply:
I think this is part of a larger effort to update the directory structure for the esp boards, since it's kind of a mess. This currently matches what we have already done previously, so I'm biased to keep as is so that the merge is clean. The task to update structure is being tracked.